### PR TITLE
Realign `phase2` with `master`

### PR DESCRIPTION
**Pull request by niklaslong**
As `master` is protected, it first needs to be merged into a clone of `phase2` because of the conflicts in #447. Supersedes #447.

### CODE REVIEW

**Comment by niklaslong** [comment]
This will need a new PR following the recent changes on `master`—will push once we've synced with @Pratyush.

File: phase1-cli/scripts/phase1_chunked.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ if [ "$PROVING_SYSTEM" == "groth16" ]; then
 else
   MAX_CHUNK_INDEX=1 # we have 2 chunks, since we have a total of 2^11-1 powers
 fi
-CURVE="bw6"
+CURVE="bls12_377"
```
Comment (Pratyush) [inline phase1-cli/scripts/phase1_chunked.sh:14]: why this change?

File: phase1-coordinator/src/coordinator.rs [hunk 2]
```diff
@@ -277,7 +278,7 @@ impl From<CoordinatorError> for anyhow::Error {
 /// for mocking system time during testing.
 pub trait TimeSource: Send + Sync {
     /// Provide the current time now in the UTC timezone
-    fn utc_now(&self) -> DateTime<Utc>;
+    fn now_utc(&self) -> OffsetDateTime;
```
Comment (Pratyush) [inline phase1-coordinator/src/coordinator.rs:281]: Is there a reason for this renaming? (Not a blocker or anything, just wondering why it was renamed)

File: phase1-coordinator/src/coordinator.rs [hunk 3]
```diff
@@ -751,15 +757,9 @@ impl Coordinator {
...
     ///
     /// Returns `true` if the given participant has finished contributing
-    /// in the current round.
     ///
     #[inline]
     pub fn is_finished_contributor(&self, participant: &Participant) -> bool {
-        // Check that the participant is a current contributor.
-        if !self.is_current_contributor(participant) {
-            return false;
-        }
```
Comment (Pratyush) [inline phase1-coordinator/src/coordinator.rs:761]: Why this change?

File: phase1-wasm/Cargo.toml [hunk 4]
```diff
@@ -12,36 +12,57 @@ edition = "2018"
...
-phase1 = { path = "../phase1" }
-setup-utils = { path = "../setup-utils" }
-snarkvm-curves = { git = "https://github.com/AleoHQ/snarkVM.git", rev = "fc997c" }
-snarkvm-fields = { git = "https://github.com/AleoHQ/snarkVM.git", rev = "fc997c" }
+http = "0.2"
+phase1 = { path = "../phase1", default-features = false }
+setup-utils = { path = "../setup-utils", default-features = false }
+setup1-shared = { path = "../setup1-shared" }
+snarkvm-dpc = { git = "https://github.com/AleoHQ/snarkVM.git", branch = "setup_wasm", default-features = false, features = [ "wasm" ], optional = true }
+snarkvm-utilities = { git = "https://github.com/AleoHQ/snarkVM.git", branch = "setup_wasm", default-features = false, optional = true }
```
Comment (Pratyush) [inline phase1-wasm/Cargo.toml:20]: Do we want to pin these also to the same commit as the other `snarkvm` dependencies?

File: phase1-wasm/src/lib.rs [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
 extern crate serde_derive;
 
-mod phase1;
+cfg_if::cfg_if! {
+    if #[cfg(not(test))] {
+        #[cfg(feature = "wasm")]
+        mod contributor;
+        #[cfg(feature = "wasm")]
+        mod errors;
+        mod pool;
```
Comment (Pratyush) [inline phase1-wasm/src/lib.rs:10]: Maybe we should move the `mod pool` outside the `cfg(feature = "wasm")` chain?

Comment (Pratyush) [reply phase1-wasm/src/lib.rs:10]: (Just to make sure it's visibly not `cfg`-ed

File: phase1-wasm/src/phase1.rs [hunk 6]
```diff
@@ -1,19 +1,16 @@
...
+
+#[cfg(not(test))]
+use setup_utils::{derive_rng_from_seed, get_rng, user_system_randomness};
+use snarkvm_curves::PairingEngine;
+
+#[cfg(not(test))]
+use snarkvm_curves::{bls12_377::Bls12_377, bw6_761::BW6_761};
 
 use rand::{CryptoRng, Rng};
 use wasm_bindgen::prelude::*;
```
Comment (Pratyush) [inline phase1-wasm/src/phase1.rs:16]:
```suggestion
```

**Comment by Pratyush** [comment]
btw which part should I focus on reviewing? not sure what's new and what's already in `master`.

**Review by Pratyush — APPROVED** [review]

File: phase1-coordinator/src/coordinator.rs (hunk 2 above)
Comment (niklaslong) [reply phase1-coordinator/src/coordinator.rs:281]: I think it's just realigning the internal api with the wrapper function there for consistency.

**Comment by niklaslong** [comment]
All of it is already in `master`, I'm merely doing the realigning. Unfortunately I'm not intimitaly familiar with all the changes, @julesdesmit would be the person to ask for some of these questions.

**Comment by Pratyush** [comment]
Sounds good, looks good to me in that case! We can probably address the comments in a separate PR. Not sure how to merge this PR since the audit checks are currently failing.

**Comment by niklaslong** [comment]
There's no way we're going to get that green at the moment 🤔 I don't have admin powers here unfortunately—@julesdesmit?

**Comment by julesdesmit** [comment]
@niklaslong i think @zosorock may have that power - otherwise I think @howardwu would be the only other one

**Comment by Pratyush** [comment]
We can be sneaky and add a commit temporarily disabling the audit check action =P

**Comment by apruden2008** [comment]
I can merge it

**Comment by Pratyush** [comment]
Do we need to merge `phase2` into master, or not yet?

**Comment by niklaslong** [comment]
That depends: do we want a single branch that contains all the code, or do we want to archive phase 1 and continue building on another branch? That latter might be safer and consistent with what we did in e.g. snarkOS/snarkVM.
